esp-hal/src/delay.rs

```diff
@@ -30,36 +30,25 @@
 //! [DelayUs]: embedded_hal_02::blocking::delay::DelayUs
 //! [embedded-hal]: https://docs.rs/embedded-hal/1.0.0/embedded_hal/delay/index.html
 
-use fugit::HertzU64;
 pub use fugit::MicrosDurationU64;
 
+use crate::clock::Clocks;
+
 /// Delay driver
 ///
 /// Uses the `SYSTIMER` peripheral internally for RISC-V devices, and the
 /// built-in Xtensa timer for Xtensa devices.
 #[derive(Clone, Copy)]
-pub struct Delay {
-    freq: HertzU64,
-}
-
-impl Delay {
-    /// Delay for the specified number of milliseconds
-    pub fn delay_millis(&self, ms: u32) {
-        for _ in 0..ms {
-            self.delay_micros(1000);
-        }
-    }
-}
+#[non_exhaustive]
+pub struct Delay {}
```

Comment:
```suggestion
pub struct Delay;
```
Is there any reason not to just make `Delay` a unit struct here?

Reply:
Not much, just wanted to be extra sure users can't construct one without `Clocks`.